pkg/scaffold/helm/chart.go

```diff
@@ -265,3 +270,116 @@ func createChartFromRemote(destDir string, opts CreateChartOptions) (*chart.Char
+	var errs []error
+	groups := map[string]map[string]struct{}{}
+	gotmplRegex := regexp.MustCompile(`{{.*}}`)
+
+	for _, tmpl := range chart.GetTemplates() {
+		data := string(tmpl.GetData())
+		name := tmpl.GetName()
+		yamlTmpls := strings.Split(data, "\n---\n")
+
+		for i, yamlTmpl := range yamlTmpls {
+			var apiVersion, kind string
+
+			clean := func(s string) string {
+				s = strings.TrimSpace(s)
+				s = strings.Trim(s, "'\"")
+				return s
+			}
+
+			// Extract the apiVersion and kind from the template
+			lines := strings.Split(yamlTmpl, "\n")
+			for _, line := range lines {
+				line = strings.TrimSpace(line)
+				if strings.HasPrefix(line, "apiVersion") {
+					apiVersion = clean(strings.TrimPrefix(line, "apiVersion:"))
+				} else if strings.HasPrefix(line, "kind:") {
+					kind = clean(strings.TrimPrefix(line, "kind:"))
+				}
+			}
+
+			// Skip templates that don't look like Kubernetes resources
+			if apiVersion == "" && kind == "" {
+				continue
+			}
+
+			// If only one of kind and apiVersion is set, we won't be able to
+			// create a valid role rule, so append an error and continue.
+			if kind == "" {
+				errs = append(errs, fmt.Errorf("could not find resource kind %s[%d]", name, i))
+				continue
+			}
+			if apiVersion == "" {
+				errs = append(errs, fmt.Errorf("could not find resource apiVersion in %s[%d]", name, i))
+				continue
+			}
+
+			// Parse the apiVersion and skip with an error if it fails.
+			gv, err := schema.ParseGroupVersion(apiVersion)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("could not parse group and version from apiVersion %s in %s[%d]: %s", apiVersion, name, i, err))
+				continue
+			}
+
+			// If the group or kind contain a go template, we won't be able to
+			// create a valid role rule, so append and error and continue.
+			group := gv.Group
+			if gotmplRegex.MatchString(group) {
+				errs = append(errs, fmt.Errorf("could not parse group (`%s`) from apiVersion in %s[%d]: group is templated", group, name, i))
+				continue
+			}
+			if gotmplRegex.MatchString(kind) {
+				errs = append(errs, fmt.Errorf("could not parse resource from kind (`%s`) in %s[%d]: kind is templated", kind, name, i))
+				continue
+			}
+
+			// Make a best guess at the resource name. It's possible for this
+			// to create an incorrect name if the kind has a custom plural form.
+			resourceName := inflect.Pluralize(strings.ToLower(kind))
+
+			if groups[gv.Group] == nil {
+				groups[gv.Group] = map[string]struct{}{}
+			}
+			groups[gv.Group][resourceName] = struct{}{}
+
+		}
+	}
+	rules := []rbacv1.PolicyRule{}
+	for group, resourceNames := range groups {
+		resources := []string{}
+		for resource := range resourceNames {
+			resources = append(resources, resource)
+		}
+		sort.Strings(resources)
+		rules = append(rules, rbacv1.PolicyRule{
+			APIGroups: []string{group},
+			Resources: resources,
+			Verbs:     []string{rbacv1.VerbAll},
```

Comment:
So if I understand correctly we are just giving it `*`, I understand we can't fine tune this, but maybe we can document that the user double checks the generated rules?

Reply:
I think `*` is reasonable for these rules because the Helm operator needs to create, get, update, and delete all of the templated resources.

However, since the code now renders the default manifest and uses that to build rules, it is possible that some resources require custom values to be set so that they are present in the release manifest. I'll add a log at the end, that says something along the lines of:

`Double check deploy/role.yaml to verify correct RBAC policy rules. The existing rules were created from the default chart manifest, so some rules may be overly broad or missing.`